engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/skipInclude/SkipIncludeTransform.kt

```diff
@@ -0,0 +1,125 @@
+package graphql.nadel.enginekt.transform.skipInclude
+
+import graphql.execution.MergedField
+import graphql.introspection.Introspection
+import graphql.language.Field
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.NadelExecutionContext
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.transform.NadelTransform
+import graphql.nadel.enginekt.transform.NadelTransformFieldResult
+import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
+import graphql.nadel.enginekt.transform.query.NadelQueryTransformer
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.nadel.enginekt.transform.skipInclude.SkipIncludeTransform.State
+import graphql.nadel.enginekt.util.resolveObjectTypes
+import graphql.nadel.enginekt.util.toBuilder
+import graphql.normalized.ExecutableNormalizedField
+import graphql.normalized.ExecutableNormalizedField.newNormalizedField
+import graphql.schema.GraphQLSchema
+
+internal class SkipIncludeTransform : NadelTransform<State> {
```

Comment:
So this transform handles the edge case where all fields are deleted. I'll update the class with docs to mention this.

Fields are automatically removed by the executable normalised factory _if_ the `@skip` or `@include` directives are true or false respectively. Because the field is entirely removed, we can end up sending an empty selection set _if_ all the fields are removed.

We obviously can't send an empty selection set, so we need to add a `__typename` selection.